diff --git a/parse.js b/parse.js
index 47a09fc..4196e9b 100644
--- a/parse.js
+++ b/parse.js
@@ -13,6 +13,11 @@ module.exports = function (config, content, file, list, logger) {
 
         if (src) {
             if (!src.match(/^http/) && !src.match(/^\/\//)) {
+                var endUrl = src.indexOf('?');
+                if (endUrl == -1) {
+                    endUrl = src.length;
+                }
+                src = src.substring(0, endUrl);
                 paths.push(fs.realpathSync(src));
 	        } else {
                 logger.debug('skipping script: ' + src);
@@ -25,10 +30,12 @@ module.exports = function (config, content, file, list, logger) {
 
     function addPathsToFileList(paths) {
         if (paths.length) {
-            paths = paths.reverse();
+            //paths = paths.reverse();
+
             var index = _.findIndex(files, {pattern: file.originalPath});
             files.splice(index, 1);
             _.each(paths, function (path) {
+                index = index +1;
                 files.splice(index, 0, {
                     pattern: path,
                     served: true,
diff --git a/test/test.js b/test/test.js
index c548634..f720c4d 100644
--- a/test/test.js
+++ b/test/test.js
@@ -10,11 +10,16 @@ describe('includelocalscripts', function () {
                         '<script src="//secondUrl.com"></script>' +
                         '<script src="./test/first.js"></script>' +
                         '<script src="test/second.js"></script>' +
+                        '<script src="./test/third.js?p1=1&p2=2"></script>' +
+                        '<script src="test/four.js?p3=3"></script>' +
                    '</head>' +
                '</html>';
 
     var baseDir = __dirname;
-    var urls = [ path.join(baseDir + '/first.js'), path.join(baseDir + '/second.js') ];
+    var urls = [ path.join(baseDir + '/first.js'),
+                 path.join(baseDir + '/second.js'), 
+                 path.join(baseDir + '/third.js'), 
+                 path.join(baseDir + '/four.js')];
     var config = { filePrefix: 'dist/' };
     var file = { originalPath: ''};
     var list = { reload: function(files) { return files; }};
